docs/rfc/0698-cli-infrastructure.adoc

```diff
@@ -0,0 +1,299 @@
+libraries. However, this leaves the question of where the binary might
+live. So far, it has been an unspoken formalism that all binaries live
+in `radicle-bins` where we can provide a `Cargo.lock` file. All
+functionality will live in `radicle-link`, including the `main`
+function, and the entry in `radicle-bins` will be a thin wrapper
+around this function.
+
+== Core
+
+[quote]
+------
+We swears to serve the master of the precious. We will swear on
+the…on the precious.
+------
+-- Gollum, The Two Towers
+
+With all the details in place, we define a non-exhaustive list of the
+core commands that we wish to include for the first iteration. More
+components MAY be added as the result of future RFCs. We SHALL keep a
+list of the core commands as part of the `radicle-link` documentation.
+
+=== `rad-identities`
+
+The `rad identities` subcommand provides the management of identity
+documents and their metadata. It is further split into subcommands
+which we sketch out below:
+
+* `project [ create | get | list | update | checkout | review ]` --
+  modify and retrieve `Project` identities.
+* `person [ create | get | list | update | checkout | review ]` --
+  modify and retrieve `Person` identities.
+* `any [ get | list ]` -- retrieve identities where the caller does not
+  know what kind of identity it is. The CLI will make a best-effort to
+  infer the type, or else return an opaque blob.
+* `local [ set | get | default ]` -- modify or retrieve a local identity
+* `tracking [ track | untrack | list ]` -- modify and list the tracking
+  graph for a given identity
+* `rad-refs [ self | signed | delegates | delegate ]` -- retrieve the
+  blobs that live under `rad/<refname>`
+* `refs [ heads | notes | tags | category ]` -- list the reference names
+  under the respective category, where `category` is a user supplied category.
+
+=== `rad-replicate`
+
+* `ask` -- tell the `daemon` that you wish to replicate the
+  given URN. Pending <<rl-141, radicle-link#141>>.
+* `replicate` -- if the peer/address are known for the given URN,
+  attempt to replicate the identity. This requires the activation of
+  the peer-to-peer daemon.
+
+=== `rad-profile`
+
+This command will provide the management of keys and the `Profile`
+construct in `librad`. The following actions will be defined:
+
+* `create`— generate a new key and a new `Profile`, which implies a
+  separate storage.
+* `get`— get the current `Profile`'s identifier
+* `set`— set the current `Profile` to the identifier
+  passed in. *Note* that this should refuse to do so if the daemon is online.
+* `list`— list the available `Profile` identifiers
+* `ssh-add`— add the current `Profile`'s key to the `ssh-agent`
+
+=== `rad-daemon`
+
+This command provides control over and introspection into the peer-to-peer daemon.
+
+* `start` -- request to start the peer-to-peer daemon.
+* `stop` -- request to stop the peer-to-peer daemon.
+* `status` -- request the status of the peer-to-peer daemon.
```

Comment:
What exactly does this return? Given that the main purpose of the daemon is the operation of the `net::Peer`, it should be assumed that the info that can be requested is closer to what can be obtained through the tincan, there is an outline of that [here](https://github.com/radicle-dev/radicle-link/blob/xla/rfc-p2p-node/docs/rfc/0696-p2p-node.adoc#requestresponse-payloads).

Reply:
Ya, so this should likely be split out into the separate things it can retrieve as outlined in #696, i.e. `peers`, `membership`, and `stats`.